nano/nano_node/entry.cpp

```diff
@@ -793,25 +795,41 @@ int main (int argc, char * const * argv)
 				}
 			}
 			// Processing blocks
-			std::cerr << boost::str (boost::format ("Starting processing %1% active blocks\n") % max_blocks);
+			std::cout << boost::str (boost::format ("Starting processing %1% blocks\n") % max_blocks);
 			auto begin (std::chrono::high_resolution_clock::now ());
 			while (!blocks.empty ())
 			{
 				auto block (blocks.front ());
 				node->process_active (block);
 				blocks.pop_front ();
 			}
+			size_t count (0);
+			while (node->ledger.cache.block_count != max_blocks + 1)
+			{
+				std::this_thread::sleep_for (std::chrono::milliseconds (10));
+				++count;
+				// Message each 15 seconds
+				if ((count % 1500) == 0)
+				{
+					std::cout << boost::str (boost::format ("%1% (%2%) blocks processed (unchecked)") % node->ledger.cache.block_count % node->ledger.cache.unchecked_count) << std::endl;
+				}
+			}
+			// Waiting for final transaction commit
 			uint64_t block_count (0);
+			{
+				auto transaction (node->store.tx_begin_read ());
+				block_count = node->store.block_count (transaction).sum ();
```

Comment:
Is this not redundant given the loop below?

Reply:
Changed